### PR TITLE
Release v0.5.0 doc validation part 2

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?
Continuation of https://github.com/ray-project/kuberay/pull/997. As in that PR,

1. This PR mainly focuses on correctness validation rather than document improvement.
2. We use the nightly KubeRay operator image to test the documents.

- [x] https://github.com/ray-project/kuberay/blob/master/docs/guidance/ingress.md
   * The script does not always work when run all at once, likely due to some race condition.  Added a troubleshooting step to unblock the user if this happens.  Ideally we should figure out the correct "wait" step to insert in the script to fix the issue.
- [x] https://github.com/ray-project/kuberay/blob/master/docs/guidance/rayjob.md
  * Passes 10/10 times on Ray 2.3 after https://github.com/ray-project/kuberay/pull/1000. 
- [x] https://github.com/ray-project/kuberay/blob/master/docs/guidance/tls.md
  * Passes. Only tested "TL;DR" path.  We should explicitly add the command for "`# Install v0.5.0 KubeRay operator`".  I used `kind create cluster` followed by `helm install kuberay-operator --set image.repository=kuberay/operator --set image.tag=nightly ~/kuberay/helm-chart/kuberay-operator`
- [x] https://github.com/ray-project/kuberay/blob/master/docs/guidance/pod-command.md
  * Passes
- [x] https://github.com/ray-project/kuberay/blob/master/docs/guidance/pod-security.md
  * Passes. Note that the job "succeeds" but the script itself raises an error:

  ```
  Traceback (most recent call last):
  File "samples/xgboost_example.py", line 28, in <module>
    result = trainer.fit()
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/train/base_trainer.py", line 363, in fit
    result_grid = tuner.fit()
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/tune/tuner.py", line 292, in fit
    return self._local_tuner.fit()
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/tune/impl/tuner_internal.py", line 455, in fit
    analysis = self._fit_internal(trainable, param_space)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/tune/impl/tuner_internal.py", line 573, in _fit_internal
    **args,
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/tune/tune.py", line 679, in run
    callbacks, sync_config, metric=metric, progress_metrics=progress_metrics
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/tune/utils/callback.py", line 105, in _create_default_callbacks
    callbacks.append(TBXLoggerCallback())
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/tune/logger/tensorboardx.py", line 165, in __init__
    from tensorboardX import SummaryWriter
  File "/home/ray/anaconda3/lib/python3.7/site-packages/tensorboardX/__init__.py", line 5, in <module>
    from .torchvis import TorchVis
  File "/home/ray/anaconda3/lib/python3.7/site-packages/tensorboardX/torchvis.py", line 10, in <module>
    from .writer import SummaryWriter
  File "/home/ray/anaconda3/lib/python3.7/site-packages/tensorboardX/writer.py", line 16, in <module>
    from .comet_utils import CometLogger
  File "/home/ray/anaconda3/lib/python3.7/site-packages/tensorboardX/comet_utils.py", line 7, in <module>
    from .summary import _clean_tag
  File "/home/ray/anaconda3/lib/python3.7/site-packages/tensorboardX/summary.py", line 12, in <module>
    from .proto.summary_pb2 import Summary
  File "/home/ray/anaconda3/lib/python3.7/site-packages/tensorboardX/proto/summary_pb2.py", line 16, in <module>
    from tensorboardX.proto import tensor_pb2 as tensorboardX_dot_proto_dot_tensor__pb2
  File "/home/ray/anaconda3/lib/python3.7/site-packages/tensorboardX/proto/tensor_pb2.py", line 16, in <module>
    from tensorboardX.proto import resource_handle_pb2 as tensorboardX_dot_proto_dot_resource__handle__pb2
  File "/home/ray/anaconda3/lib/python3.7/site-packages/tensorboardX/proto/resource_handle_pb2.py", line 42, in <module>
    serialized_options=None, file=DESCRIPTOR),
  File "/home/ray/anaconda3/lib/python3.7/site-packages/google/protobuf/descriptor.py", line 561, in __new__
    _message.Message._CheckCalledFromGeneratedFile()
```
TypeError: Descriptors cannot not be created directly.
If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
If you cannot immediately regenerate your protos, some other possible workarounds are:
 1. Downgrade the protobuf package to 3.20.x or lower.
 2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower)
 ```

 this doesn't affect the main point of the doc, which is to demonstrate security settings.
<!-- Please give a short summary of the change and the problem this solves. -->

## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: docs/guidance/ingress.md [hunk 1]
```diff
@@ -201,3 +201,12 @@ kubectl describe ingress raycluster-ingress-head-ingress
 #        [Note] The forward slash at the end of the address is necessary. `<ip>/raycluster-ingress`
 #               will report "404 Not Found".
 ```
+
```
Comment (kevin85421) [inline docs/guidance/ingress.md:204]: Maybe we just need to add a comment in Step 2 that says to execute the kubectl wait command after the Pod for the NGINX ingress controller is created or just `sleep 10`. This is an expected behavior for `kubectl wait`.

**Comment by kevin85421** [comment]
For `pod-security.md`, this bug is totally from Ray Docker images. It has been fixed by Ray 2.2 but happened again in Ray 2.3 images. See https://github.com/ray-project/kuberay/issues/873 for more details.

**Review by kevin85421 — COMMENTED** [review]
Remove the section "Example: NGINX Ingress on KinD (built-in ingress support)" in [ingress.md](https://github.com/ray-project/kuberay/blob/master/docs/guidance/ingress.md) and remove https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/ray-cluster.ingress.yaml.

**Comment by gvspraveen** [comment]
For 
> does not gracefully handle multiple jobs with the same submission ID submitted in quick succession
May be in the description here link to your [PR in 2.4 which fixes this](https://github.com/ray-project/ray/pull/33259)

**Comment by architkulkarni** [comment]
> For
> 
> > does not gracefully handle multiple jobs with the same submission ID submitted in quick succession
> > May be in the description here link to your [PR in 2.4 which fixes this](https://github.com/ray-project/ray/pull/33259)

Actually it's works now after Kai-Hsun's fix https://github.com/ray-project/kuberay/pull/1000, I've updated the PR description

File: docs/guidance/ingress.md (hunk 1 above)
Comment (kevin85421) [reply docs/guidance/ingress.md:204]: The commands are copied from [here](https://kind.sigs.k8s.io/docs/user/ingress/#ingress-nginx).

**Comment by architkulkarni** [comment]
> For `pod-security.md`, this bug is totally from Ray Docker images. It has been fixed by Ray 2.2 but happened again in Ray 2.3 images. See #873 for more details. Maybe we should update the example script in Step 5.2.

I confirmed that this script passes with `rayproject/ray-ml:nightly`, so I've updated the pod security sample YAML to use `ray-ml:latest`.  When the Ray 2.4 docker images are released, the test should pass.

**Comment by architkulkarni** [comment]
Discussed offline, we'll use Ray 2.2.0 for the pod security doc for now until we can confirm ray:latest works (likely after 2.4.0 release of Ray)

**Review by kevin85421 — APPROVED** [review]
LGTM

File: docs/guidance/ingress.md [hunk 2]
```diff
@@ -168,6 +100,7 @@ EOF
 
 # Step 2: Install NGINX ingress controller
 kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/main/deploy/static/provider/kind/deploy.yaml
+sleep 10
```
Comment (architkulkarni) [inline docs/guidance/ingress.md:103]:
```suggestion
sleep 10 # Wait for the Kubernetes API Server to create the related resources
```

**Pull request merged**